From aa83f3eaea29d6e5d562fa08b47443a82fe3382a Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Fri, 2 Dec 2016 17:34:55 -0600
Subject: [PATCH 136/268] sodium_init() needs its return value checked

I have fixed the one call to sodium_init().  I also imported the
warn_unused_result from my system's sodium/core.h to avoid this going
unnoticed if future calls to sodium_init() are added.
---
 libsodium/include/sodium/core.h | 3 ++-
 ntpd/ntpd.c                     | 5 ++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/libsodium/include/sodium/core.h b/libsodium/include/sodium/core.h
index 7faa983..3ca4476 100644
--- a/libsodium/include/sodium/core.h
+++ b/libsodium/include/sodium/core.h
@@ -9,7 +9,8 @@ extern "C" {
 #endif
 
 SODIUM_EXPORT
-int sodium_init(void);
+int sodium_init(void)
+            __attribute__ ((warn_unused_result));
 
 #ifdef __cplusplus
 }
diff --git a/ntpd/ntpd.c b/ntpd/ntpd.c
index 3d58081..1fd5ef7 100644
--- a/ntpd/ntpd.c
+++ b/ntpd/ntpd.c
@@ -663,7 +663,10 @@ ntpdmain(
         /*
 	 * Initialize libsodium and its RNG
 	 */
-	sodium_init();
+	if (sodium_init() < 0) {
+		msyslog(LOG_ERR, "sodium_init() failed");
+		exit(1);
+	}
 
 	/*
 	 * Set up signals we pay attention to locally.
-- 
2.7.4

